app/controllers/sites/new.js

```diff
@@ -3,48 +3,117 @@ import { inject as service } from '@ember/service';
 import { combineFullAddress } from 'frontend-contactgegevens-loket/models/address';
 import { tracked } from '@glimmer/tracking';
 import { task } from 'ember-concurrency';
+import { action } from '@ember/object';
+import {
+  errorValidation,
+  warningValidation,
+} from '../../validations/site-validation';
 
 export default class CreateSitesNewController extends Controller {
   @service router;
   @service store;
   @tracked isPrimarySite = false;
+  @tracked validationErrors = {};
+  @tracked validationWarnings = {};
+  @tracked showWarningModal = false;
+
+  @action
+  reset() {
+    this.isPrimarySite = false;
+    this.validationErrors = {};
+    this.validationWarnings = {};
+    this.showWarningModal = false;
+  }
 
   get isLoading() {
     return this.saveTask.isRunning || this.cancelTask.isRunning;
   }
 
+  async validateData() {
+    const { address, primaryContact, secondaryContact, site } = this.model;
+    const validationData = {
+      siteType: site.siteType.get('label'),
+      street: address.street,
+      country: address.country,
+      number: address.number,
+      postcode: address.postcode,
+      municipality: address.municipality,
+      province: address.province,
+      fullAddress: address.fullAddress,
+      telephonePrimary: primaryContact.telephone
+        ? primaryContact.telephone.replace(/ /g, '')
+        : '',
+      emailPrimary: primaryContact.email,
+      websitePrimary: primaryContact.website,
+      telephoneSecondary: secondaryContact.telephone,
+    };
+
+    const errorValidationResult = errorValidation.validate(validationData);
+    const warningValidationResult = warningValidation.validate(validationData);
+    return {
+      errors: errorValidationResult.error,
+      warnings: warningValidationResult.error,
+    };
+  }
+
   saveTask = task(async (event) => {
```

Comment:
This name is not correct. It should be 'validateAndSave' task. You have to split them up. Only use tasks when necassary.
Build a structure like this
* Action: handleAddClick (user clicks 'voeg toe'): Run the validation functions and if no errors and no warnings run the save task. If errors return. If warnings show dialog.
* Action: handleModalOkClick (user clicks OK in de modal dialog): Close the dialog box and run the save task.
* Async function save
* saveTask: task wrapping the save function.